File: src/main/java/ml/wonwoo/springdatashowcase/order/OrderInitializerData.java
package ml.wonwoo.springdatashowcase.order;

import lombok.RequiredArgsConstructor;
import ml.wonwoo.springdatashowcase.data.InitializerData;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class OrderInitializerData implements InitializerData {

    private final OrderRepository orderRepository;

    @Override
    public void initialize() {
        Order order = orderRepository.save(new Order());
        order.orderPaid();
        orderRepository.save(order);
    }
}

File: kotlin/src/main/kotlin/ml/wonwoo/springdatashowcase/order/OrderInitializerData.kt
package ml.wonwoo.springdatashowcase.order

import ml.wonwoo.springdatashowcase.data.InitializerData
import org.springframework.stereotype.Component

@Component
class OrderInitializerData(private val orderRepository: OrderRepository)
    : InitializerData {

    override fun initialize() {

        val order = orderRepository.save(Order())

        order.orderPaid()

        orderRepository.save(order)

    }
}